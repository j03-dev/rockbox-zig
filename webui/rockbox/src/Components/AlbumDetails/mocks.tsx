export const tracks = [
  {
    id: "1",
    trackNumber: 1,
    title: "Gasoline (Live)",
    artist: "Seether",
    time: "2:57",
  },
  {
    id: "2",
    trackNumber: 2,
    title: "Driven Under (Live)",
    artist: "Seether",
    time: "4:58",
  },
  {
    id: "3",
    trackNumber: 3,
    title: "Diseased (Live)",
    artist: "Seether",
    time: "3:46",
  },
  {
    id: "4",
    trackNumber: 4,
    title: "Truth (Live)",
    artist: "Seether",
    time: "5:15",
  },
  {
    id: "5",
    trackNumber: 5,
    title: "Immortality (Live)",
    artist: "Seether",
    time: "5:02",
  },
  {
    id: "6",
    trackNumber: 6,
    title: "Tied My Hands (Live)",
    artist: "Seether",
    time: "5:16",
  },
  {
    id: "7",
    trackNumber: 7,
    title: "Sympathetic (Live)",
    artist: "Seether",
    time: "4:12",
  },
  {
    id: "8",
    trackNumber: 8,
    title: "Fine Again (Live)",
    artist: "Seether",
    time: "5:05",
  },
  {
    id: "9",
    trackNumber: 9,
    title: "Broken (Live)",
    artist: "Seether",
    time: "4:17",
  },
  {
    id: "10",
    trackNumber: 10,
    title: "The Gift (Live)",
    artist: "Seether",
    time: "5:36",
  },
  {
    id: "11",
    trackNumber: 11,
    title: "Remedy (Live)",
    artist: "Seether",
    time: "3:40",
  },
  {
    id: "12",
    trackNumber: 12,
    title: "Plastci Man (Live)",
    artist: "Seether",
    time: "3:34",
  },
  {
    id: "13",
    trackNumber: 13,
    title: "The Gist (Alternate Mix)",
    artist: "Seether",
    time: "3:24",
  },
];
