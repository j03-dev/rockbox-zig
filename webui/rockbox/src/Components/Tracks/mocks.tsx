export const tracks = [
  {
    id: "1",
    trackNumber: 1,
    title: "Changed Plan",
    artist: "Waveshaper",
    album: "Forgotten Shapes",
    time: "2:39",
    albumArt:
      "https://resources.tidal.com/images/5b33d07f/7d28/417d/8f8f/86291d0b6b34/80x80.jpg",
    albumId: "1",
    artistId: "1",
  },
  {
    id: "2",
    trackNumber: 2,
    title: "Judas Mind",
    artist: "Seether",
    album: "The Surface Seems So Far",
    time: "4:34",
    albumArt:
      "https://resources.tidal.com/images/0a30cfd9/5b9b/4fd4/b172/a4c625ce1dbb/80x80.jpg",
    albumId: "2",
    artistId: "2",
  },
  {
    id: "3",
    trackNumber: 3,
    title: "Oh My (Remix)",
    artist: "Roscoe Dash",
    album: "Dash Effect",
    time: "2:46",
    albumArt:
      "https://resources.tidal.com/images/d6630bea/6cc0/4d08/89b7/138e49bf8d0f/80x80.jpg",
    albumId: "3",
    artistId: "3",
  },
];
