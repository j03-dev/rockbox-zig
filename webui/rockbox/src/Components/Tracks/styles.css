table {
  border-spacing: 0;
}

.album-cover-container.songs {
  height: 48px;
  justify-content: center;
  align-items: center;
  display: flex;
  position: relative;
  cursor: pointer;
  margin-right: 10px;
}

.album-cover-container.songs .floating-play {
  display: none;
  position: absolute;
  left: 17px;
  top: 10px;
 }

.album-cover-container.songs:hover .floating-play {
  display: block;
}

.album-cover-container.songs:hover img {
  opacity: 0.4;
 }