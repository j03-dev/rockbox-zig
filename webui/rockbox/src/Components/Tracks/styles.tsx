import styled from "@emotion/styled";
import { Link as RouterLink } from "react-router-dom";

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100%;
`;

export const MainView = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
`;

export const Title = styled.div`
  font-size: 24px;
  font-family: RockfordSansMedium;
  margin-bottom: 20px;
`;

export const IconButton = styled.button`
  background-color: transparent;
  cursor: pointer;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  &:hover {
    opacity: 0.6;
  }
`;

export const Hover = styled.button`
  color: transparent;
  background-color: transparent;
  border: none;
  opacity: 1 !important;
  cursor: pointer;
  &:hover,
  &:focus {
    color: #000;
    opacity: 1 !important;
  }
`;

export const ButtonGroup = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const ContentWrapper = styled.div`
  overflow-y: auto;
  height: calc(100vh - 60px);
  padding-left: 20px;
  padding-right: 20px;
`;

export const AlbumCover = styled.img`
  height: 48px;
  width: 48px;
`;

export const FilterContainer = styled.div`
  margin-top: 30px;
  margin-bottom: 40px;
`;

export const BackButton = styled.button`
  border: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 30px;
  width: 30px;
  border-radius: 15px;
  background-color: #f7f7f8;
  margin-top: 26px;
  margin-bottom: 46px;
  position: absolute;
  z-index: 1;
`;

export const Link = styled(RouterLink)`
  color: #000;
  text-decoration: none;
  font-family: RockfordSansRegular;
  &:hover {
    text-decoration: underline;
  }
`;
