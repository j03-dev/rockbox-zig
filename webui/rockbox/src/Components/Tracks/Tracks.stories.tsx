import type { Meta, StoryObj } from "@storybook/react";

import Tracks from "./Tracks";
import { tracks } from "./mocks";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Components/Tracks",
  component: Tracks,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
} satisfies Meta<typeof Tracks>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Default: Story = {
  args: {
    tracks,
  },
};
