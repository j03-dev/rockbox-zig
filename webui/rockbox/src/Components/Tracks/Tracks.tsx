/* eslint-disable @typescript-eslint/no-explicit-any */
import { FC } from "react";
import { createColumnHelper } from "@tanstack/react-table";
import Sidebar from "../Sidebar";
import ControlBar from "../ControlBar";
import {
  AlbumCover,
  ButtonGroup,
  Container,
  ContentWrapper,
  FilterContainer,
  Hover,
  IconButton,
  Link,
  MainView,
  Title,
} from "./styles";
import { EllipsisHorizontal } from "@styled-icons/ionicons-sharp";
import Add from "../Icons/Add";
import HeartOutline from "../Icons/HeartOutline";
import { Track } from "../../Types/track";
import Table from "../VirtualizedTable";
import { tracks } from "./mocks";
import Filter from "../Filter";
import "./styles.css";

const columnHelper = createColumnHelper<Track>();
const columns = [
  columnHelper.accessor("trackNumber", {
    header: "#",
    size: 20,
    cell: (info) => (
      <div
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
        }}
      >
        {info.getValue()}
      </div>
    ),
  }),
  columnHelper.accessor("albumArt", {
    header: "Title",
    size: 54,
    cell: (info) => <AlbumCover src={info.getValue()} alt="album art" />,
  }),
  columnHelper.accessor("title", {
    header: "",
    cell: (info) => (
      <div
        style={{
          minWidth: 150,
          width: "calc(100% - 20px)",
          fontSize: 14,
          textOverflow: "ellipsis",
          overflow: "hidden",
          whiteSpace: "nowrap",
          cursor: "pointer",
          color: "#000",
        }}
      >
        {info.getValue()}
      </div>
    ),
  }),
  columnHelper.accessor("artist", {
    header: "Artist",
    cell: (info) => (
      <div
        style={{
          minWidth: 150,
          width: "calc(100% - 20px)",
          fontSize: 14,
          textOverflow: "ellipsis",
          overflow: "hidden",
          whiteSpace: "nowrap",
          cursor: "pointer",
          color: "#000",
        }}
      >
        <Link href="#">{info.getValue()}</Link>
      </div>
    ),
  }),
  columnHelper.accessor("album", {
    header: "Album",
    cell: (info) => (
      <div
        style={{
          minWidth: 150,
          width: "calc(100% - 20px)",
          fontSize: 14,
          textOverflow: "ellipsis",
          overflow: "hidden",
          whiteSpace: "nowrap",
          cursor: "pointer",
          color: "#000",
        }}
      >
        <Link href="#">{info.getValue()}</Link>
      </div>
    ),
  }),
  columnHelper.accessor("time", {
    header: "Time",
    size: 50,
    cell: (info) => info.getValue(),
  }),
  columnHelper.accessor("id", {
    header: "",
    size: 100,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    cell: (_info) => (
      <ButtonGroup style={{ justifyContent: "flex-end", alignItems: "center" }}>
        <IconButton>
          <Hover>
            <EllipsisHorizontal size={24} />
          </Hover>
        </IconButton>
        <IconButton>
          <Add color="#000" size={24} />
        </IconButton>
        <IconButton>
          <HeartOutline color="#000" />
        </IconButton>
      </ButtonGroup>
    ),
  }),
];

const Tracks: FC = () => {
  return (
    <Container>
      <Sidebar active="songs" />
      <MainView>
        <ControlBar />
        <ContentWrapper>
          <Title>Songs</Title>
          <FilterContainer>
            <Filter placeholder="Search song" onChange={() => {}} />
          </FilterContainer>
          <Table columns={columns as any} tracks={tracks} />
        </ContentWrapper>
      </MainView>
    </Container>
  );
};

export default Tracks;
