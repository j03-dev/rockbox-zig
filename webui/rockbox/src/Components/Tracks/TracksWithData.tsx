import { FC, useEffect, useState } from "react";
import Tracks from "./Tracks";
import { useTracksQuery } from "../../Hooks/GraphQL";
import { useTimeFormat } from "../../Hooks/useFormat";
import { Track } from "../../Types/track";

const TracksWithData: FC = () => {
  const { data, loading } = useTracksQuery();
  const [tracks, setTracks] = useState<Track[]>([]);
  const { formatTime } = useTimeFormat();

  useEffect(() => {
    if (!data || loading) {
      return;
    }

    setTracks(
      data.tracks.map((x, i) => ({
        id: x.id!,
        trackNumber: i + 1,
        title: x.title,
        artist: x.artist,
        album: x.album,
        time: formatTime(x.length),
        albumArt: x.albumArt
          ? `http://localhost:6062/covers/${x.albumArt}`
          : undefined,
        albumId: x.albumId,
        artistId: x.artistId,
      }))
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data, loading]);

  const onPlayTrack = (trackId: string) => {
    console.log(">>", trackId);
  };

  return (
    <>
      {tracks.length > 0 && (
        <Tracks tracks={tracks} onPlayTrack={onPlayTrack} />
      )}
    </>
  );
};

export default TracksWithData;
