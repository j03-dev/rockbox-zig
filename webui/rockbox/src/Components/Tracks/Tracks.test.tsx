import { render } from "@testing-library/react";
import Tracks from "./Tracks";
import Providers from "../../Providers";
import { tracks } from "./mocks";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import { mocks } from "../../mocks";
import { vi } from "vitest";

describe("Tracks", () => {
  it("should render", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Providers>
            <Tracks tracks={tracks} onPlayTrack={vi.fn()} />
          </Providers>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
