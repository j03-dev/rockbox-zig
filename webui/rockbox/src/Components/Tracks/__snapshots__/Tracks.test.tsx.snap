// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tracks > should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="css-e1989k"
    >
      <div
        class="css-1tlxqlf"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <img
            alt="Rockbox"
            src="/src/Assets/rockbox-icon.svg"
            style="width: 40px; margin-bottom: 20px; margin-left: 12px;"
          />
        </a>
        <a
          class="css-tujm8o"
          color="initial"
          href="/albums"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
            color="initial"
            fill="currentColor"
            focusable="false"
            height="20"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
            />
            <path
              d="M12 8a4 4 0 1 0 4 4 4 4 0 0 0-4-4zm0 6a2 2 0 1 1 2-2 2 2 0 0 1-2 2z"
            />
          </svg>
          <div>
            Albums
          </div>
        </a>
        <a
          class="css-tujm8o"
          color="initial"
          href="/artists"
        >
          <svg
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4.22a5.67 5.67 0 0 0-9.68 4.57l-8 9.79 3.3 3.3 9.79-8c.18 0 .36.05.55.05a5.7 5.7 0 0 0 4-9.73ZM5.74 19.86l-1.38-1.38 6.44-7.89a5.48 5.48 0 0 0 2.83 2.84Zm13.21-8.65a4.2 4.2 0 1 1 0-5.94 4.17 4.17 0 0 1 0 5.95Z"
              fill="initial"
            />
          </svg>
          <div
            style="margin-left: 6px;"
          >
            Artists
          </div>
        </a>
        <a
          class="css-8v05qj"
          color="#fe099c"
          href="/tracks"
        >
          <svg
            height="20"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.1 4.65v11.26a3.45 3.45 0 1 0 1.5 2.84V5.85l10.2-2.36v10.62A3.45 3.45 0 1 0 21.3 17V1.61Zm-2 16a2 2 0 1 1 2-2 2 2 0 0 1-1.95 2.05Zm11.7-1.8a1.95 1.95 0 1 1 2-1.85 2 2 0 0 1-1.95 1.9Z"
              fill="#fe099c"
            />
          </svg>
          <div
            style="margin-left: 6px;"
          >
            Songs
          </div>
        </a>
        <a
          class="css-tujm8o"
          color="initial"
          href="/files"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
            color="initial"
            fill="none"
            focusable="false"
            height="19"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            style="margin-right: 6px;"
            viewBox="0 0 24 24"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="22"
              x2="2"
              y1="12"
              y2="12"
            />
            <path
              d="M5.45 5.11 2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
            />
            <line
              x1="6"
              x2="6.01"
              y1="16"
              y2="16"
            />
            <line
              x1="10"
              x2="10.01"
              y1="16"
              y2="16"
            />
          </svg>
          <div>
            Files
          </div>
        </a>
      </div>
      <div
        class="css-y9r6ap"
      >
        <div
          class="css-14bvc2y"
        >
          <div
            class="css-1h69jy4"
          >
            <div
              class="css-1ytum11"
            >
              <button
                class="css-10asy2d"
              >
                <svg
                  data-test="icon-PlaybackShuffle"
                  fill="none"
                  height="16"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16.2 7h2.85l-1.3 1.3c-.4.4-.4 1 0 1.4.25.2.5.3.75.3s.5-.1.7-.3l3-3 .15-.15.05-.05c0-.025.013-.037.025-.05.012-.013.025-.025.025-.05 0-.021 0-.033.004-.044.005-.015.017-.027.046-.056 0-.021 0-.033.004-.044.005-.015.017-.027.046-.056.05-.15.05-.25 0-.4 0-.05 0-.05-.05-.1 0-.021 0-.033-.004-.044-.005-.015-.017-.027-.046-.056 0-.025-.012-.037-.025-.05-.012-.013-.025-.025-.025-.05 0 0 0-.05-.05-.05l-.15-.15-3-3c-.4-.4-1-.4-1.4 0-.4.4-.4 1 0 1.4L19.1 5h-2.85c-1.75 0-3.3.9-4.25 2.35L6.85 15.6c-.55.9-1.5 1.4-2.55 1.4H2.5c-.55 0-1 .45-1 1s.45 1 1 1h1.8c1.75 0 3.3-.9 4.25-2.35L13.7 8.4c.5-.85 1.5-1.4 2.5-1.4ZM2.5 7h1.8c1.05 0 2 .5 2.55 1.4l.45.75 1.2-1.9C7.55 5.8 6 4.95 4.3 4.95H2.5c-.55 0-1 .45-1 1 0 .6.45 1.05 1 1.05Zm19.95 11.2v.05c0 .05 0 .1-.05.1v.05c-.05.1-.1.2-.2.3l-3 3c-.2.2-.45.3-.7.3-.25 0-.5-.1-.75-.3-.4-.4-.4-1 0-1.4l1.3-1.3H16.2c-1.7 0-3.25-.85-4.2-2.3l1.25-1.9.45.75c.55.9 1.5 1.4 2.55 1.4h2.85l-1.3-1.3c-.4-.4-.4-1 0-1.4.4-.4 1-.4 1.4 0l3 3c.1.1.15.2.2.3v.05c.05.05.05.1.05.15v.05c.05.15.05.25 0 .4Z"
                    fill="#000"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
              <button
                class="css-10asy2d"
              >
                <svg
                  fill="none"
                  height="18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="ionicon"
                    d="M3.55 2c.304 0 .55.243.55.542V7.16l8.001-4.716a1.224 1.224 0 0 1 1.23-.015c.413.23.669.677.669 1.162v9.818c0 .486-.256.932-.669 1.162-.382.216-.853.21-1.23-.015L4.1 9.84v4.618a.546.546 0 0 1-.55.542.546.546 0 0 1-.55-.542V2.542C3 2.243 3.246 2 3.55 2Z"
                    style="fill: #000;"
                  />
                </svg>
              </button>
              <button
                class="css-10asy2d"
              >
                <svg
                  fill="none"
                  height="32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.657 27a2.842 2.842 0 0 1-1.258-.292C8.536 26.283 8 25.458 8 24.563V6.438c0-.899.536-1.721 1.399-2.146a2.856 2.856 0 0 1 2.571.028l17.817 9.273c.755.411 1.213 1.131 1.213 1.906 0 .774-.458 1.495-1.213 1.906l-17.82 9.275a2.846 2.846 0 0 1-1.31.32Z"
                    style="fill: #000;"
                  />
                </svg>
              </button>
              <button
                class="css-10asy2d"
              >
                <svg
                  fill="none"
                  height="18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="ionicon"
                    d="M14.45 2a.546.546 0 0 0-.55.542V7.16L5.899 2.444a1.224 1.224 0 0 0-1.23-.015C4.256 2.659 4 3.105 4 3.591v9.818c0 .486.256.932.669 1.162.382.216.853.21 1.23-.015L13.9 9.84v4.618c0 .299.246.542.55.542.304 0 .55-.243.55-.542V2.542A.546.546 0 0 0 14.45 2Z"
                    style="fill: #000;"
                  />
                </svg>
              </button>
              <button
                class="css-10asy2d"
              >
                <svg
                  data-test="icon-PlaybackInactiveRepeat"
                  fill="none"
                  height="16"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M16.293 1.293a1 1 0 0 1 1.414 0l3 3a1 1 0 0 1 0 1.414l-3 3a1 1 0 1 1-1.414-1.414L17.586 6H8.5c-3.298 0-6 2.702-6 6a1 1 0 1 1-2 0c0-4.402 3.598-8 8-8h9.086l-1.293-1.293a1 1 0 0 1 0-1.414ZM23.5 12a1 1 0 1 0-2 0c0 3.298-2.702 6-6 6H6.414l1.293-1.293a1 1 0 1 0-1.414-1.414l-3 3a1 1 0 0 0 0 1.414l3 3a1 1 0 0 0 1.414-1.414L6.414 20H15.5c4.402 0 8-3.598 8-8Z"
                    fill="#000"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="css-frud38"
          >
            <div
              class="css-4ijxew"
            >
              <svg
                height="28"
                viewBox="0 0 24 24"
                width="28"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.1 4.65v11.26a3.45 3.45 0 1 0 1.5 2.84V5.85l10.2-2.36v10.62A3.45 3.45 0 1 0 21.3 17V1.61Zm-2 16a2 2 0 1 1 2-2 2 2 0 0 1-1.95 2.05Zm11.7-1.8a1.95 1.95 0 1 1 2-1.85 2 2 0 0 1-1.95 1.9Z"
                  fill="#b1b2b5"
                />
              </svg>
            </div>
            <div
              class="css-rbpzuh"
            >
              <div
                style="color: rgb(177, 178, 181); text-align: center;"
              >
                No track playing
              </div>
            </div>
          </div>
          <div
            class="css-nb4c1y"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="css-10asy2d"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
                fill="currentColor"
                focusable="false"
                height="21"
                viewBox="0 0 20 20"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.4 9H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1h5.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm2 5H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1h7.8c.552 0 .6-.447.6-1s-.048-1-.6-1zM8.6 6h7.8c.552 0 .6-.447.6-1s-.048-1-.6-1H8.6c-.552 0-.6.447-.6 1s.048 1 .6 1zM5.4 9H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm0 5H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1zm0-10H3.6c-.552 0-.6.447-.6 1s.048 1 .6 1h1.8c.552 0 .6-.447.6-1s-.048-1-.6-1z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-18cz59b"
        >
          <div
            class="css-b8ko1a"
          >
            Songs
          </div>
          <div
            class="css-9wcbnz"
          >
            <div
              class="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8"
              data-baseweb="input"
            >
              <div
                class="af b9 ba bb ar as ax ay az b0 b7 bc bd b5 be bf"
              >
                <svg
                  data-test="icon-ActionActiveSearch"
                  fill="none"
                  height="20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4 9.5a5.5 5.5 0 1 1 11 0 5.5 5.5 0 0 1-11 0ZM9.5 2a7.5 7.5 0 1 0 4.549 13.463l2.744 2.744a1 1 0 0 0 1.414-1.414l-2.744-2.744A7.5 7.5 0 0 0 9.5 2Z"
                    fill="#ACAAAA"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <div
                class="af bg bh ar as ax ay az b0 bd b5"
                data-baseweb="base-input"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  autocomplete="on"
                  class="ae b5 bi bj bk bl bm bn bo bp bq bg br bs bt bu bv bw b6 bx ax by az b0 bz c0 c1"
                  inputmode="text"
                  name="filter"
                  placeholder="Search song"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            style="margin-bottom: 60px;"
          >
            <div
              style="height: 96px;"
            >
              <table
                style="width: 100%;"
              >
                <thead>
                  <tr
                    style="height: 36px; color: rgba(0, 0, 0, 0.54);"
                  >
                    <th
                      style="text-align: left; width: 20px;"
                    >
                      #
                    </th>
                    <th
                      style="text-align: left; width: 54px;"
                    >
                      Title
                    </th>
                    <th
                      style="text-align: left; width: 150px;"
                    />
                    <th
                      style="text-align: left; width: 150px;"
                    >
                      Artist
                    </th>
                    <th
                      style="text-align: left; width: 150px;"
                    >
                      Album
                    </th>
                    <th
                      style="text-align: left; width: 50px;"
                    >
                      Time
                    </th>
                    <th
                      style="text-align: left; width: 100px;"
                    />
                  </tr>
                </thead>
                <tbody />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
